### PR TITLE
Add cap_array_value function

### PR DESCRIPTION
**Pull request by toniecrumley**
The cap_array_value function will retrieve a value from an array file based on it's zero based index.  If no index is provided the Slurm array environment variable will be used by default.
Setup:
```
cd ~/bin/capture
git checkout main
git pull
git checkout cap-array-value
git submodule update --init --recursive

```
Create a test project:
```
cap new hello
cd hello

```
Create an array file `data/person_list.array`:
```
Tabea
Lizzy
Tonie
```
Create a test job `src/01_hello.sh`:
```
#!/bin/bash

#SBATCH --nodes=1
#SBATCH --ntasks=1
#SBATCH --mem-per-cpu=8G
#SBATCH --cpus-per-task=1
#SBATCH --time=00:05:00
#SBATCH --partition=express
#SBATCH --array=0-2

person=$(cap_array_value "$CAP_DATA_PATH/person_list.array")
echo "Hello $person!!"
```
Run the test job:
```
cap run src/01_hello.sh

```
View the test job output:
```
cat logs/*

```
Test other scenarios...

Restore to released CAPTURE after testing:
```
cap update

```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -292,12 +292,27 @@ v0.0.3
...
+```
+- `FILE` The file containing an array value on each line.
+- `INDEX` The optional zero based index for the value of the array. If a value
+    is not provided then SLURM_ARRAY_TASK_ID environment variable will be used
+    as the default.
+
+The folowing example will retrieve an array value based on the Slurm environment
+variable default index.
+```
+sample=$(cap_array_value "$CAP_DATA_PATH/sample_list.array")
```
Comment (tsoelter) [inline README.md:308]: An example of how to use/set `INDEX` would also be helpful.

File: README.md [hunk 2]
```diff
@@ -292,12 +292,27 @@ v0.0.3
...
 # Helper Functions
+## cap_array_value
+Retrieves a value from an array file based on a zero based index.
+```
+cap_arrary_value FILE [INDEX]
+```
+- `FILE` The file containing an array value on each line.
+- `INDEX` The optional zero based index for the value of the array. If a value
+    is not provided then SLURM_ARRAY_TASK_ID environment variable will be used
+    as the default.
```
Comment (tsoelter) [inline README.md:303]: I am unsure if I am using `INDEX` correctly. I modified the testing script by setting `INDEX` to 1 to see what happens, but when I do that, the same "sample" (aka person's name, in this case "Lizzy") gets used for each array job. Is this how it's intended to function?

**Review by tsoelter — CHANGES_REQUESTED** [review]

File: README.md (hunk 2 above)
Comment (emmafjones) [reply README.md:303]: yes, I think that's how it works

**Review by emmafjones — DISMISSED** [review]
this worked for me, I was able to increase the array size and add my name to get expected outputs

File: README.md (hunk 2 above)
Comment (tsoelter) [reply README.md:303]: I discussed this with Tonie offline and got clarification.

File: lib/functions/cap_array_value.sh [hunk 3]
```diff
@@ -0,0 +1,30 @@
+#!/bin/bash
+
+cap_array_value() {
+  # Check that the required file name parameter was provided
+  if [ "$#" -lt 1 ]; then
+    echo "Error: incorrect number of parameters" >&2
+    echo "Usage: cap_array_value FILE [INDEX]" >&2
```
[the remark this answers is not in the document]
Comment (toniecrumley) [reply lib/functions/cap_array_value.sh:7]: More specific information was added to the error message.

**Comment by toniecrumley** [comment]
@tsoelter @emmafjones Code review comments have been addressed.  FYI, Emma, the changes I pushed in response to code review undid your approval.

**Review by emmafjones — APPROVED** [review]

**Review by tsoelter — APPROVED** [review]

**Pull request merged**